tensorflow/core/common_runtime/executor.cc

```diff
@@ -2156,6 +2159,7 @@ void ExecutorState::PropagateOutputs(const TaggedNode& tagged_node,
     if (output_frame != nullptr) {
       // This is the case when node is not Enter, Exit, or NextIteration.
       DCHECK(input_frame == output_frame);
+      mutex_lock l(output_frame->mu);
```

Comment:
we already hold lock on `input_frame` and since the two alias, this line would deadlock